src/Nethermind/Nethermind.Specs/ChainSpecStyle/ChainSpecBasedSpecProvider.cs

```diff
@@ -222,19 +223,55 @@ public void UpdateMergeTransitionInfo(long? blockNumber, UInt256? terminalTotalD
 
         public IReleaseSpec GenesisSpec => _transitions.Length == 0 ? null : _transitions[0].Release;
 
-        public IReleaseSpec GetSpec(ForkActivation forkActivation) =>
-                _transitions.TryGetSearchedItem(forkActivation,
+        public IReleaseSpec GetSpec(ForkActivation activation)
+        {
+            if (activation.Timestamp is null)
+            {
+                return _transitions.TryGetSearchedItem(activation,
                     CompareTransitionOnBlock,
-                    out (ForkActivation, ReleaseSpec Release) transition)
+                    out (ForkActivation Activation, ReleaseSpec Release) transition)
                     ? transition.Release
-                    : null;
+                    : GenesisSpec;
+            }
+            List<(ForkActivation Activation, IReleaseSpec Release)> filterdByBlockNumber = new();
```

Comment:
Do you think it is possible to avoid this allocation?
GetSpec is a bit complicated now
Do we have unit tests for GetSpec?